### PR TITLE
[AUR-243] data store test batch 8

### PR DESCRIPTION
**Pull request by Ficzer**

### CODE REVIEW

File: requirements/DATA_STORE_AURORA_0005/testing.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+**1. Steps:**
+
+* Build the project with "make" command
+* Run ./work/binaries/demo file
+
+**2. Expected results:**
+
+* Project should succesfully build
+* The running binary should output logs in a loop with item reception messages
```
Comment (Lurkerpas) [inline requirements/DATA_STORE_AURORA_0005/testing.md]: Requirement is about updates

Comment (rphh) [reply requirements/DATA_STORE_AURORA_0005/testing.md]: Add an example expected output

File: requirements/DATA_STORE_AURORA_0006/testing.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+**1. Steps:**
+
+* Build the project with "make" command
+* Run ./work/binaries/demo file
+
+**2. Expected results:**
+
+* Project should succesfully build
+* The running binary should output logs in a loop with item reception messages
```
Comment (rphh) [inline requirements/DATA_STORE_AURORA_0006/testing.md]: requirement is  The Data Interface shall provide the capability to delete data in memory
please add an example output

File: requirements/DATA_STORE_AURORA_0008/testing.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+**1. Steps:**
+
+* Build the project with "make" command
+* Run ./work/binaries/demo file
+
+**2. Expected results:**
+
+* Project should succesfully build
+* The running binary should output logs in a loop with item reception messages
```
Comment (rphh) [inline requirements/DATA_STORE_AURORA_0008/testing.md]: Please provide an expected output

File: requirements/DATA_STORE_AURORA_0007/review-of-design.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Review of design DATA_STORE_AURORA_0007
```
Comment (rphh) [inline requirements/DATA_STORE_AURORA_0007/review-of-design.md:1]: So, this does not require a model, right?

File: requirements/DATA_STORE_AURORA_0008/work/client1/C/src/client1.c [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+static asn1SccT_EventRetrieveLogMessage log_message;
+
+static int counter;
+static int subscribed;
+static int key;
+static asn1SccUShortInteger logIndex;
+
+static void print_log(asn1SccT_EventRetrieveLogMessage log_message)
+{
+    switch(log_message.u.log_item.operation.kind)
```
Comment (rphh) [inline requirements/DATA_STORE_AURORA_0008/work/client1/C/src/client1.c]: check the valu eof  `log_message.kind`

File: requirements/DATA_STORE_AURORA_0005/testing.md (hunk 1 above)
Comment (Ficzer) [reply requirements/DATA_STORE_AURORA_0005/testing.md]: Done

File: requirements/DATA_STORE_AURORA_0008/work/client1/C/src/client1.c [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    }
+    else if(counter < 12 && counter % 4 == 2)
+    {
+        delete_request.item_key = key;
+        client1_RI_Delete(&delete_request);
+        ++counter;
+    }
+    else if(counter < 12 && counter % 4 == 3)
+    {
+        client1_RI_RetrieveLogItem(&log_message, &logIndex);
```
Comment (rphh) [inline requirements/DATA_STORE_AURORA_0008/work/client1/C/src/client1.c:107]: is it possible to add check for more than one entry?

File: requirements/DATA_STORE_AURORA_0005/work/client1/C/src/client1.c [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    }
+}
+
+void client1_PI_notifyRetrieve(const asn1SccT_EventRetrieveMessage* ev)
+{
+    switch(ev->kind)
+    {
+        case T_EventRetrieveMessage_item_retrieved_PRESENT:
+            printf("client1: item retrieved key = %lu\n", ev->u.item_retrieved.item_key);
+            printf("client1: item value coefficient = %lu\n", ev->u.item_retrieved.item_value.u.coefficient);
```
Comment (rphh) [inline requirements/DATA_STORE_AURORA_0005/work/client1/C/src/client1.c]: check value of `ev->u.item_retrieved.item_value.kind`

**Review by rphh — DISMISSED** [review]

File: requirements/DATA_STORE_AURORA_0006/testing.md (hunk 2 above)
Comment (Ficzer) [reply requirements/DATA_STORE_AURORA_0006/testing.md]: Done

File: requirements/DATA_STORE_AURORA_0008/testing.md (hunk 3 above)
Comment (Ficzer) [reply requirements/DATA_STORE_AURORA_0008/testing.md]: Done

File: requirements/DATA_STORE_AURORA_0007/review-of-design.md (hunk 4 above)
Comment (Ficzer) [reply requirements/DATA_STORE_AURORA_0007/review-of-design.md:1]: It does not

File: requirements/DATA_STORE_AURORA_0008/work/client1/C/src/client1.c (hunk 5 above)
Comment (Ficzer) [reply requirements/DATA_STORE_AURORA_0008/work/client1/C/src/client1.c]: Done

File: requirements/DATA_STORE_AURORA_0008/work/client1/C/src/client1.c (hunk 6 above)
Comment (Ficzer) [reply requirements/DATA_STORE_AURORA_0008/work/client1/C/src/client1.c:107]: Done

File: requirements/DATA_STORE_AURORA_0005/work/client1/C/src/client1.c (hunk 7 above)
Comment (Ficzer) [reply requirements/DATA_STORE_AURORA_0005/work/client1/C/src/client1.c]: Done

File: requirements/DATA_STORE_AURORA_0007/review-of-design.md (hunk 4 above)
Comment (Lurkerpas) [reply requirements/DATA_STORE_AURORA_0007/review-of-design.md:1]: Yes, validation method is D, description was discussed on teams.

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**